forge-ai/src/main/java/forge/ai/ComputerUtilCost.java

```diff
@@ -598,6 +598,20 @@ public boolean apply(Card card) {
             }
         }
 
+        // Bail early on Casualty in case there are no cards that would make sense to pay with
+        if (sa.getHostCard().hasKeyword(Keyword.CASUALTY)) {
```

Comment:
there is already some logic in `PlayerControllerAi.chooseNumberForKeywordCost`, so that might be a better placement

Reply:
I tried that, but couldn't make it work :/ If I return 0 from there, it happens too late and the AI has already made the decision to go for the spell ability with the Casualty cost applied, which will result in a "AI failed to play..." misplay.

Reply:
hmn ok :/